packages/flutter_blue/tizen/src/flutter_blue_tizen_plugin.cc

```diff
@@ -0,0 +1,314 @@
+       *
+       */
+
+      result->Success(flutter::EncodableValue(NULL));
+
+    } else if (method_call.method_name() == "state") {
+      result->Success(
+          flutter::EncodableValue(flutter_blue_tizen::MessageToVector(
+              bluetooth_manager_->BluetoothState())));
+
+    } else if (method_call.method_name() == "isOn") {
+      result->Success(flutter::EncodableValue(
+          (bluetooth_manager_->BluetoothState().state() ==
+           proto::gen::BluetoothState_State::BluetoothState_State_ON)));
+
+    } else if (method_call.method_name() == "startScan") {
+      proto::gen::ScanSettings scan_settings;
+      std::vector<uint8_t> encoded = std::get<std::vector<uint8_t>>(args);
+
+      try {
+        scan_settings.ParseFromArray(encoded.data(), encoded.size());
+        bluetooth_manager_->StartBluetoothDeviceScanLE(scan_settings);
+        result->Success(flutter::EncodableValue(NULL));
+      } catch (const std::exception& e) {
+        result->Error(e.what());
+      }
+
+    } else if (method_call.method_name() == "stopScan") {
+      try {
+        bluetooth_manager_->StopBluetoothDeviceScanLE();
+        result->Success(flutter::EncodableValue(NULL));
+      } catch (const std::exception& e) {
+        result->Error(e.what());
+      }
+
+    } else if (method_call.method_name() == "getConnectedDevices") {
+      proto::gen::ConnectedDevicesResponse response;
+      auto bluetooth_proto_devices =
+          bluetooth_manager_->GetConnectedProtoBluetoothDevices();
+
+      for (auto& device : bluetooth_proto_devices) {
+        *response.add_devices() = std::move(device);
+      }
+
+      result->Success(flutter::EncodableValue(
+          flutter_blue_tizen::MessageToVector(response)));
+
+    } else if (method_call.method_name() == "connect") {
+      std::vector<uint8_t> encoded = std::get<std::vector<uint8_t>>(args);
+      proto::gen::ConnectRequest connect_request;
+
+      try {
+        connect_request.ParseFromArray(encoded.data(), encoded.size());
+        bluetooth_manager_->Connect(connect_request);
+        result->Success(flutter::EncodableValue(NULL));
+      } catch (const std::exception& e) {
+        result->Error(e.what());
+      }
+
+    } else if (method_call.method_name() == "disconnect") {
+      std::string device_id = std::get<std::string>(args);
+      try {
+        bluetooth_manager_->Disconnect(device_id);
+        result->Success(flutter::EncodableValue(NULL));
+      } catch (const std::exception& e) {
+        result->Error(e.what());
+      }
+
+    } else if (method_call.method_name() == "deviceState") {
+      std::string device_id = std::get<std::string>(args);
+      std::scoped_lock lock(bluetooth_manager_->bluetoothDevices().mutex_);
+      auto it = bluetooth_manager_->bluetoothDevices().var_.find(device_id);
+
+      if (it != bluetooth_manager_->bluetoothDevices().var_.end()) {
+        auto& device = it->second;
```

Comment:
This logic should be handled by `BluetoothManager` by something like `GetBluetoothDevice`. It's exposing internal member and how it's being used.